src/test/ui/suggestions/missing-trait-bounds-for-method-call.stderr

```diff
@@ -0,0 +1,39 @@
+error[E0599]: no method named `foo` found for reference `&Foo<T>` in the current scope
+  --> $DIR/missing-trait-bounds-for-method-call.rs:14:14
+   |
+LL | struct Foo<T> {
+   | ------------- doesn't satisfy `Foo<T>: Bar`
+...
+LL |         self.foo();
+   |              ^^^ method not found in `&Foo<T>`
+   |
+   = note: the method `foo` exists but the following trait bounds were not satisfied:
+           `T: Bar`
+           which is required by `Foo<T>: Bar`
+           `T: std::default::Default`
+           which is required by `Foo<T>: Bar`
+help: consider restricting the type parameters to satisfy the obligations
+   |
+LL | struct Foo<T> where T: Bar, T: std::default::Default {
+   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error[E0599]: no method named `foo` found for reference `&Fin<T>` in the current scope
+  --> $DIR/missing-trait-bounds-for-method-call.rs:27:14
+   |
+LL | struct Fin<T> where T: Bar {
+   | -------------------------- doesn't satisfy `Fin<T>: Bar`
+...
+LL |         self.foo();
+   |              ^^^ method not found in `&Fin<T>`
+   |
+   = note: the method `foo` exists but the following trait bounds were not satisfied:
+           `T: std::default::Default`
+           which is required by `Fin<T>: Bar`
+help: consider restricting the type parameter to satisfy the obligation
```

Comment:
consider rewording `obligations` to `requirements`? as `obligation` isn't a common term outside the internal rustc.

Reply:
What about "consider restricting the type parameter to satisfy the trait bound"?